dot/sync/tip_syncer_test.go

```diff
@@ -0,0 +1,404 @@
+			}
+			assert.Equal(t, tt.want, got)
+		})
+	}
+}
+
+func Test_tipSyncer_handleTick(t *testing.T) {
+	t.Parallel()
+
+	type fields struct {
+		blockStateBuilder    func(ctrl *gomock.Controller) BlockState
+		pendingBlocksBuilder func(ctrl *gomock.Controller) DisjointBlockSet
+		readyBlocks          *blockQueue
+	}
+	tests := map[string]struct {
+		fields fields
+		want   []*worker
+		err    error
+	}{
+		"base case": {
+			fields: fields{
+				pendingBlocksBuilder: func(ctrl *gomock.Controller) DisjointBlockSet {
+					mockDisjointBlockSet := NewMockDisjointBlockSet(ctrl)
+					mockDisjointBlockSet.EXPECT().size().Return(1).Times(2)
+					mockDisjointBlockSet.EXPECT().getBlocks().Return([]*pendingBlock{
+						{number: 2},
+						{number: 3},
+						{number: 4,
+							header: &types.Header{
+								Number: 4,
+							},
+						},
+						{number: 5,
+							header: &types.Header{
+								Number: 5,
+							},
+							body: &types.Body{},
+						},
+					})
+					mockDisjointBlockSet.EXPECT().removeBlock(common.Hash{})
+					return mockDisjointBlockSet
+				},
+				blockStateBuilder: func(ctrl *gomock.Controller) BlockState {
+					mockBlockState := NewMockBlockState(ctrl)
+					mockBlockState.EXPECT().GetHighestFinalisedHeader().Return(&types.Header{
+						Number: 2,
+					}, nil)
+					mockBlockState.EXPECT().HasHeader(common.Hash{}).Return(false, nil)
+					return mockBlockState
+				},
+				readyBlocks: newBlockQueue(3),
+			},
+			want: []*worker{
+				{
+					startNumber:  uintPtr(3),
+					targetNumber: uintPtr(2),
+					targetHash: common.Hash{5, 189, 204, 69, 79, 96, 160, 141, 66, 125, 5, 231, 241,
+						159, 36, 15, 220, 57, 31, 87, 10, 183, 111, 203, 150, 236, 202, 11, 88, 35, 211, 191},
+					pendingBlock: &pendingBlock{number: 3},
+					requestData:  bootstrapRequestData,
+					direction:    network.Descending,
+				},
+				{
+					startNumber:  uintPtr(4),
+					targetNumber: uintPtr(4),
+					pendingBlock: &pendingBlock{
+						number: 4,
+						header: &types.Header{
+							Number: 4,
+						},
+					},
+					requestData: network.RequestedDataBody + network.RequestedDataJustification,
+				},
+				{
+					startNumber:  uintPtr(4),
+					targetNumber: uintPtr(2),
+					direction:    network.Descending,
+					pendingBlock: &pendingBlock{
+						number: 5,
+						header: &types.Header{
+							Number: 5,
+						},
+						body: &types.Body{},
+					},
+					requestData: bootstrapRequestData,
+				},
+			},
+			err: nil,
+		},
+	}
+	for name, tt := range tests {
+		tt := tt
+		ctrl := gomock.NewController(t)
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+			s := &tipSyncer{
+				blockState:    tt.fields.blockStateBuilder(ctrl),
+				pendingBlocks: tt.fields.pendingBlocksBuilder(ctrl),
+				readyBlocks:   tt.fields.readyBlocks,
+			}
+			got, err := s.handleTick()
+			if tt.err != nil {
+				assert.EqualError(t, err, tt.err.Error())
+			} else {
+				assert.NoError(t, err)
+			}
+			assert.Equal(t, tt.want, got)
```

Comment:
Also assert the entire tipSyncer since handleTick does modify its internal state.

Reply:
No need for it, tipSyncer isn't modified